### PR TITLE
[Refactor] Made notification system

### PR DESCRIPTION
**Pull request by YashKamboj**
Signed-off-by: Yash Kamboj <yashkamboj29@gmail.com>

**Description** 
Made notification system
Screenshot (135)

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Please add refactor and ui labels.

File: ui/components/Notification/MesheryNotification.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    Tab,
+  } from "@mui/material";
+  import NotificationsIcon from '@mui/icons-material/Notifications';
+  import DoneAllIcon from '@mui/icons-material/DoneAll';
+  import MesheryEventViewer from "./MesheryEventViewer";
+
+   
+  function MesheryNotification () {
+
+  const DrawerButton = styled(IconButton)((theme) => ({
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesheryNotification.js:20]: Before creating a new button, ask yourself whether this button can be derived from an existing one or whether this button has any similarities with the existing ones. Try to reuse components and styles as much as possible.

File: ui/components/Notification/MesheryNotification.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    padding : '0.45rem',
+    margin : '0.2rem',
+    backgroundColor : '#000000',
+    color : '#FFFFFF',
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesheryNotification.js:24]: Let's follow a rule -> always define colors in the color palette and refer them

File: ui/components/Notification/MesheryNotification.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  } from "@mui/material";
+  import NotificationsIcon from '@mui/icons-material/Notifications';
+  import DoneAllIcon from '@mui/icons-material/DoneAll';
+  import MesheryEventViewer from "./MesheryEventViewer";
+
+   
+  function MesheryNotification () {
+
+  const DrawerButton = styled(IconButton)((theme) => ({
+    padding : '0.45rem',
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesheryNotification.js:21]: use `theme.spacing()`

File: ui/components/Notification/MesheryNotification.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    margin : '0.2rem',
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesheryNotification.js:22]: use theme.spacing()

File: ui/components/Notification/MesheryNotification.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    backgroundColor : '#000000',
+    color : '#FFFFFF',
+    "&:hover" : { backgroundColor : '#FFFFFF',
+      color : '#000000' }
+  }))
+
+    const listTop = styled('div')(() => ({
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesheryNotification.js:29]: please use mui containers and grids.

File: ui/components/Notification/MesheryNotification.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+              }
+            }}
+            variant="persistent"
+            anchor="right"
+            open={open}
+            onClose={handleClose}
+          >
+            <div>
+              <div >
+                <div style={{      display : 'grid',
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesheryNotification.js:120]: not encouraged. Please use mui containers and grids.

File: ui/components/Notification/MesheryEventViewer.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import MesherySnackbarWrapper from "./MesherySnackbarWrapper"
+
+
+
+function MesheryEventViewer () {
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesheryEventViewer.js:5]: let's get into the habit of adding a short description as to what this component is supposed to do just above it's definition

File: ui/components/Notification/MesheryNotification.js [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+    return(
+        <div>
+        <IconButton color="inherit" size="large"  onClick={handleToggle}>
+        <Badge badgeContent={getNotificationCount(events)} >
+                <NotificationsIcon   />
+                </Badge>  
+            </IconButton>
+                   
+             <Drawer
+             sx={{
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesheryNotification.js:97]: again, think if this sort of styling can be reused and if so, then make it a reusable component.

File: ui/components/Notification/MesherySnackbarWrapper.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+import {
+    IconButton,
+    Typography,
+    Paper,
+    Collapse,
+    Card,
+    CardActions,
+    Grid
+  } from "@mui/material";   
+  import { SnackbarContent } from 'notistack';
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesherySnackbarWrapper.js:12]: let's move away from notistack and just use mui snackbar.

**Review by sudo-NithishKarthik — CHANGES_REQUESTED** [review]
Really good work @YashKamboj 😄 . I've added some comments.

File: ui/components/Notification/MesheryEventViewer.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import MesherySnackbarWrapper from "./MesherySnackbarWrapper"
+
+
+//Component for the  event(notification) snackbar
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesheryEventViewer.js:4]: Let us stay consistent with the way we add comments. For any declarations, we shall use multiline comments( `/* */` ) and leverage JsDoc to give additional type information.

File: ui/components/Notification/MesherySnackbarWrapper.js (hunk 9 above)
Comment (sudo-NithishKarthik) [reply ui/components/Notification/MesherySnackbarWrapper.js:12]: @YashKamboj ^^

File: ui/components/Notification/MesherySnackbarWrapper.js [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+//     success : CheckCircleIcon,
+//     warning : WarningIcon,
+//     error : ErrorIcon,
+//     info : InfoIcon,
+//   };
+
+
+  function MesherySnackbarWrapper ({ message, onClose, variant, details}) {
+    
+    // const Icon = variantIcon[variant];
```
Comment (sudo-NithishKarthik) [inline ui/components/Notification/MesherySnackbarWrapper.js:24]: let's remove unwanted comments

**Comment by leecalcote** [comment]
Merge conflict.

File: ui/components/Notification/MesherySnackbarWrapper.js (hunk 9 above)
Comment (Chadha93) [reply ui/components/Notification/MesherySnackbarWrapper.js:12]: @YashKamboj Please update this too.

Comment (YashKamboj) [reply ui/components/Notification/MesherySnackbarWrapper.js]: Done!!

**Pull request merged**